textattack/augmentation/augmenter.py

```diff
@@ -11,40 +15,100 @@ class Augmenter:
-        return [t.clean_text() for t in transformations]
+        all_transformations = set()
+        for _ in range(self.transformations_per_example):
+            indices_to_modify = set(range(len(tokenized_text.words)))
+            next_tokenized_text = tokenized_text
+            for __ in range(self.words_to_swap):
+                transformations = []
+                for constraint in self.pre_transformation_constraints:
+                    indices_to_modify = list(set(indices_to_modify) & constraint(tokenized_text, self))
+                while not len(transformations):
+                    # Loop until we find a valid transformation.
+                    if not len(indices_to_modify):
+                        # This occurs when we couldn't find valid transformations
+                        # – either the constraints were too strict, or the number
+                        # of words to swap was too high, or we were just plain 
+                        # unlucky. In any event, don't throw an error, and just 
+                        # don't return anything.
+                        break
+                    replacement_index = random.choice(indices_to_modify)
+                    indices_to_modify.remove(replacement_index)
+                    transformations = self.transformation(next_tokenized_text, indices_to_modify=[replacement_index])
+                    # Get rid of transformations we already have
+                    transformations = [t for t in transformations if t not in all_transformations]
+                    # Filter out transformations that don't match the constraints.
+                    transformations = self._filter_transformations(tokenized_text, transformations)
+                if len(transformations):
+                    next_tokenized_text = random.choice(transformations)
+            all_transformations.add(next_tokenized_text)
+        return [t.clean_text() for t in all_transformations]
     
     def augment_many(self, text_list):
         """
         Returns all possible augmentations of a list of strings according to
         `self.transformation`.
+    
+        Args:
+            text_list (list(string)): a list of strings for data augmentation
+            
+        Returns a list(string) of augmented texts.
         """
         return [self.augment(text) for text in text_list]
         
+    def augment_text_and_ids(self, text_list, id_list, show_progress=True):
+        """ Supplements a list of text with more text data. Returns the augmented
+            text along with the corresponding IDs for each augmented example.
+        """
+        if len(text_list) != len(id_list):
+            raise ValueError('List of text must be same length as list of IDs')
+        if self.transformations_per_example == 0:
+            return text_list, id_list
+        all_text_list = []
+        all_id_list = []
+        if show_progress:
```

Comment:
Why a progress bar for this but not for augment_many?

Reply:
`augment_many` returns augmentations for a single input, and I can't think of a case where you'd want so many augmentations of a single input that a progress bar is really necessary. This, on the other hand, augments an entire dataset, which can take a lot of time